extensions/amp-lightbox/0.1/amp-lightbox.js

```diff
@@ -189,36 +190,54 @@ class AmpLightbox extends AMP.BaseElement {
         .then(() => this.finalizeOpen_());
   }
 
+  /**
+   * Any child of the lightbox with the autofocus attribute should be focused
+   * after the lightbox opens.
+   * @private
+   */
+  handleAutofocus_() {
+    const autofocusElement = this.container_.querySelector('[autofocus]');
+    if (autofocusElement) {
+      tryFocus(autofocusElement);
+    }
+  }
+
+  /**
+   * @private
+   */
   finalizeOpen_() {
     if (this.isScrollable_) {
       st.setStyle(this.element, 'webkitOverflowScrolling', 'touch');
     }
-    this.mutateElement(() => {
-      st.setStyles(this.element, {
-        display: '',
-        opacity: 0,
-        // TODO(dvoytenko): use new animations support instead.
-        transition: 'opacity 0.1s ease-in',
-      });
-      Services.vsyncFor(this.win).mutate(() => {
-        st.setStyle(this.element, 'opacity', '');
-      });
-    }).then(() => {
-      const container = dev().assertElement(this.container_);
-      if (!this.isScrollable_) {
-        this.updateInViewport(container, true);
-      } else {
-        this.scrollHandler_();
-        this.updateChildrenInViewport_(this.pos_, this.pos_);
-      }
-      // TODO: instead of laying out children all at once, layout children based
-      // on visibility.
-      this.element.addEventListener('transitionend', this.boundReschedule_);
-      this.element.addEventListener('animationend', this.boundReschedule_);
-      this.scheduleLayout(container);
-      this.scheduleResume(container);
-      this.triggerEvent_(LightboxEvents.OPEN);
+
+    // This should be in a mutateElement block, but iOS focus doesn't work
+    // if triggered asynchronously.
+    st.setStyles(this.element, {
+      display: '',
+      opacity: 0,
+      // TODO(dvoytenko): use new animations support instead.
+      transition: 'opacity 0.1s ease-in',
     });
+    Services.vsyncFor(this.win).mutate(() => {
+      st.setStyle(this.element, 'opacity', '');
+    });
+
+    this.handleAutofocus_();
```

Comment:
The issue is basically that we need the `tryFocus()` to be applied to something visible, BUT due to iOS restrictions on focus needing to be synchronously called inside a click handler, we can't have it inside or waiting for a `mutateElement`. Would love to know if there's a better solution for this.